weave-js/src/components/Panel2/pyBoardGen.tsx

```diff
@@ -11,13 +11,44 @@ import {
   absoluteTargetMutation,
   makeMutation,
   useClientContext,
+  useNodeValue,
   useRefreshAllNodes,
 } from '@wandb/weave/react';
-import {useCallback} from 'react';
+import {useCallback, useMemo} from 'react';
 
 import {usePanelContext} from './PanelContext';
 import moment from 'moment';
 
+export const useBoardGeneratorsForNode = (
+  node: Node,
+  allowConfig: boolean = false
+): {
+  loading: boolean;
+  result: Array<{
+    display_name: string;
+    description: string;
+    op_name: string;
+  }>;
+} => {
+  const genBoardsNode = callOpVeryUnsafe(
+    'py_board-get_board_templates_for_node',
+    {
+      input_node: node,
+    }
+  );
+  const res = useNodeValue(genBoardsNode as any);
+  return useMemo(() => {
+    if (res.loading) {
+      return {loading: true, result: []};
+    } else {
+      return {
+        loading: false,
+        result: res.result.filter((x: any) => allowConfig || !x.config),
```

Comment:
I've noticed that there are a lot of `any`s in weave code, it would be good if we could figure out how to lock types down more.

Reply:
i properly typed it in next commit. 
```
  const res: {
    loading: boolean;
    result: Array<{
      display_name: string;
      description: string;
      op_name: string;
      config: Type;
    }>;
  } = useNodeValue(genBoardsNode as any);
```

Reply:
note: this is normally handled by the useNodeValue function, but since the incoming node is unsafe, we don't have the TS typing.